test/runtests.jl

```diff
@@ -9,7 +9,15 @@ Attributes = Dict{Symbol, Any}
     ys = ys_val * m
     ys_ret = recipe!(attr, ys)
     @test ys_ret ≈ ys_val
-    @test attr[:yguide] == string(m)
+    @test attr[:yguide] == "m"
+
+    attr = Attributes(:yguide => P"hello")
+    recipe!(attr, ys)
+    @test attr[:yguide].content == "hello"
```

Comment:
```suggestion
    @test attr[:yguide] == UnitfulRecipes.ProtectedString("hello")
```
So it does not accidentally pick up the `ys` unit.